robot_sdk/SDK/owl_sim_sdk/robot.py

```
@@ -0,0 +1,389 @@
+            rclpy.init()
+            node = Node("owl_sim_sdk")
+        self.node = node
+        
+        # Dynamic import of robot module
+        try:
+            robot_module_path = f"robogpt_tools.robot_sdk.SDK.owl_sim_sdk.robots.{robot_name}"
+            self.robot = importlib.import_module(robot_module_path)
+            self.node.get_logger().info(f"Successfully loaded robot module: {robot_name}")
+        except ImportError as e:
+            self.node.get_logger().error(f"Failed to import robot module '{robot_name}': {e}")
+            raise ImportError(f"Robot module '{robot_name}' not found. Available robots should be in robogpt_tools.robot_sdk.SDK.owl_sim_sdk.robots.{robot_name}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)